app/src/main/java/com/example/next_show/data/RecommendationClient.java

```diff
@@ -38,59 +40,48 @@ public RecommendationClient(Shows showsObj) {
         this.showsObj = showsObj;
     }
 
-    public void fetchRecommendedShows(List<String> savedShows, FeedFragment.ShowCallback showCallback, FeedFragment.GenreMatchedCallback gmCallback) {
-        this.genreMatchedCallback = gmCallback;
-        this.showCallback = showCallback;
-
-        if (savedShows.isEmpty()){
-            // if no shows to compare, call user genre matched ones
-            fetchGenreMatchedShows(showsObj, genreMatchedCallback);
+    public void fetchRelatedShows(List<String> savedShows, ResponseCallback callback) {
+        // check if no saved shows and immediately fail
+        if(savedShows.isEmpty() || savedShows == null) {
+            callback.onFailure(NO_CALL);
```

Comment:
Another option here: you could have the caller of this API first check if there are saved shows.  And then you can assert in here or do a preconditions.checkArgument that savedShows is not null and is not empty